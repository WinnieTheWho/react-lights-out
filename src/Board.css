.Board {
  margin: 0 auto;
}

h1 {
  color: yellow;
}