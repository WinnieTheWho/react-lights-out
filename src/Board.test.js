import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Board from './Board';

it('renders without crashing', () => {
  let wrapper = mount(<Board />);
  wrapper.setState({
    board: [[false, false, false, true, false],
    [false, false, false, true, false], [false, false, false, true, false],
    [false, false, false, true, false], [false, false, false, true, false]]
  });
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it('handles cell-clicks correctly', () => {
  let wrapper = mount(<Board />);
  wrapper.setState({
    board: [[false, false, false, false, false],
    [false, false, false, false, false], [false, false, false, false, false],
    [false, false, false, false, false], [false, false, false, false, false]]
  });
  wrapper.find('td').first()
    .simulate('click', { target: { value: false } });
  expect(wrapper.state().board).toEqual([[true, true, false, false, false],
  [true, false, false, false, false], [false, false, false, false, false],
  [false, false, false, false, false], [false, false, false, false, false]]);
});