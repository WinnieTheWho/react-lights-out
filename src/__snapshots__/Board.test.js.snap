// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Board
  chanceLightStartsOn={0.3}
  ncols={5}
  nrows={5}
>
  <div>
    <h1
      className="title"
    >
       Lights Out 
    </h1>
    <p
      className="direction"
    >
       The goal of the game is to switch all the lights off (gray). 
    </p>
    <p
      className="direction"
    >
       Pressing any of the lights will toggle it and the adjacent lights 
    </p>
    <table>
      <tbody>
        <tr
          key="0"
        >
          <Cell
            coords="0-0"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="0"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="0-1"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="1"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="0-2"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="2"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="0-3"
            flipCellsAroundMe={[Function]}
            isLit={true}
            key="3"
          >
            <td
              className="Cell Cell-lit"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="0-4"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="4"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
        </tr>
        <tr
          key="1"
        >
          <Cell
            coords="1-0"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="0"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="1-1"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="1"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="1-2"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="2"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="1-3"
            flipCellsAroundMe={[Function]}
            isLit={true}
            key="3"
          >
            <td
              className="Cell Cell-lit"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="1-4"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="4"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
        </tr>
        <tr
          key="2"
        >
          <Cell
            coords="2-0"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="0"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="2-1"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="1"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="2-2"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="2"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="2-3"
            flipCellsAroundMe={[Function]}
            isLit={true}
            key="3"
          >
            <td
              className="Cell Cell-lit"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="2-4"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="4"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
        </tr>
        <tr
          key="3"
        >
          <Cell
            coords="3-0"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="0"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="3-1"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="1"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="3-2"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="2"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="3-3"
            flipCellsAroundMe={[Function]}
            isLit={true}
            key="3"
          >
            <td
              className="Cell Cell-lit"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="3-4"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="4"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
        </tr>
        <tr
          key="4"
        >
          <Cell
            coords="4-0"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="0"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="4-1"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="1"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="4-2"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="2"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="4-3"
            flipCellsAroundMe={[Function]}
            isLit={true}
            key="3"
          >
            <td
              className="Cell Cell-lit"
              onClick={[Function]}
            />
          </Cell>
          <Cell
            coords="4-4"
            flipCellsAroundMe={[Function]}
            isLit={false}
            key="4"
          >
            <td
              className="Cell"
              onClick={[Function]}
            />
          </Cell>
        </tr>
      </tbody>
    </table>
  </div>
</Board>
`;
